azure-kusto-ingest/tests/test_kusto_ingest_client.py

```diff
@@ -156,6 +191,36 @@ def test_sanity_ingest_from_file(self, mock_uuid, mock_put_message_in_queue, moc
 
         assert type(upload_blob_kwargs["data"]) == io.BytesIO
 
+    @responses.activate
+    def test_ingest_from_file_wrong_endpoint(self):
+        responses.add_callback(
+            responses.POST, "https://somecluster.kusto.windows.net/v1/rest/mgmt", callback=request_error_callback, content_type="application/json"
+        )
+
+        ingest_client = QueuedIngestClient("https://somecluster.kusto.windows.net")
+        ingestion_properties = IngestionProperties(database="database", table="table", data_format=DataFormat.CSV)
+
+        current_dir = os.getcwd()
+        path_parts = ["azure-kusto-ingest", "tests", "input", "dataset.csv"]
+        missing_path_parts = []
+        for path_part in path_parts:
+            if path_part not in current_dir:
+                missing_path_parts.append(path_part)
+
+        file_path = os.path.join(current_dir, *missing_path_parts)
+
+        reached_exception = False
+        try:
+            ingest_client.ingest_from_file(file_path, ingestion_properties=ingestion_properties)
+        except Exception as e:
+            reached_exception = True
+            assert e.__class__ == KustoClientError
```

Comment:
isinstance()

Reply:
Also, why not just catch KustoClientError explicitly

Reply:
I changed this code to use self.assertRaises anyway, but just to call out my approach for catching that Exception (so you don't think I'm an idiot - what I wrote here does indeed look stupid :-) ):
When I tested this code by integrating in sample.py, its exception handling behavior was unexpected: If the integrating code used a try/except with anything other than "Exception" itself, the execution would stop after the exception was raised and reached the integrating code. But if I wanted it to continue executing after the "except", I had to specifically catch "Exception". I looked it up, and someone on StackOverflow came to that conclusion as well.

But I still may be an idiot, because I might've been wrong and instead there might've been another issue at play :-)

Reply:
try:
   do something 
except KustoClientError:
   this is ok
except exception: 
   this test has failed, but the exception is swallowed

Reply:
According to my tests, it works similarly, but with 1 slight difference:
try:
  do something
except KustoClientError:
  this is ok, **but if we get here execution will not proceed to "next_step()"**
except exception:
  this test has failed, but the exception is swallowed, **and if we get here execution will proceed to "next_step()"**
next_step()

Reply:
Then don't catch Exception class

Reply:
I don't - that's outdated code since Ariel mentioned the assertRaises